crypto/verificationhelper/qrcode.go

```diff
@@ -0,0 +1,93 @@
+package verificationhelper
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+
+	"go.mau.fi/util/random"
+
+	"maunium.net/go/mautrix/id"
+)
+
+var (
+	ErrInvalidQRCodeHeader  = errors.New("invalid QR code header")
+	ErrUnknownQRCodeVersion = errors.New("invalid QR code version")
+	ErrInvalidQRCodeMode    = errors.New("invalid QR code mode")
+)
+
+type QRCodeMode byte
+
+const (
+	QRCodeModeCrossSigning                    QRCodeMode = 0x00
+	QRCodeModeSelfVerifyingMasterKeyTrusted   QRCodeMode = 0x01
+	QRCodeModeSelfVerifyingMasterKeyUntrusted QRCodeMode = 0x02
+)
+
+type QRCode struct {
+	Mode          QRCodeMode
+	TransactionID id.VerificationTransactionID
+	Key1, Key2    [32]byte
+	SharedSecret  []byte
+}
+
+func NewQRCode(mode QRCodeMode, txnID id.VerificationTransactionID, key1, key2 [32]byte) *QRCode {
+	return &QRCode{
+		Mode:          mode,
+		TransactionID: txnID,
+		Key1:          key1,
+		Key2:          key2,
+		SharedSecret:  random.Bytes(32),
```

Comment:
Do other clients use such long shared secrets? The spec says "about 8 bytes long"